store/storetest/reaction_store.go

```diff
@@ -422,3 +424,65 @@ func testReactionBulkGetForPosts(t *testing.T, ss store.Store) {
 	require.False(t, post4IdFound, "Wrong reaction returned")
 
 }
+
+// testReactionDeadlock is a best-case attempt to recreate the deadlock scenario.
+// It atleast deadlocks 2 times out of 5.
+func testReactionDeadlock(t *testing.T, ss store.Store) {
+	post, err := ss.Post().Save(&model.Post{
+		ChannelId: model.NewId(),
+		UserId:    model.NewId(),
+	})
+	require.Nil(t, err)
+
+	reaction1 := &model.Reaction{
+		UserId:    model.NewId(),
+		PostId:    post.Id,
+		EmojiName: model.NewId(),
+	}
+	_, err = ss.Reaction().Save(reaction1)
+	require.Nil(t, err)
+
+	// different user
+	reaction2 := &model.Reaction{
+		UserId:    model.NewId(),
+		PostId:    reaction1.PostId,
+		EmojiName: reaction1.EmojiName,
+	}
+	_, err = ss.Reaction().Save(reaction2)
+	require.Nil(t, err)
+
+	// different post
+	reaction3 := &model.Reaction{
+		UserId:    reaction1.UserId,
+		PostId:    model.NewId(),
+		EmojiName: reaction1.EmojiName,
+	}
+	_, err = ss.Reaction().Save(reaction3)
+	require.Nil(t, err)
+
+	// different emoji
+	reaction4 := &model.Reaction{
+		UserId:    reaction1.UserId,
+		PostId:    reaction1.PostId,
+		EmojiName: model.NewId(),
+	}
+	_, err = ss.Reaction().Save(reaction4)
+	require.Nil(t, err)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
```

Comment:
nit: `wg.Add(1)` on two separate lines instead?

Reply:
I prefer the earlier pattern actually. :) If we know before-hand how many goroutines are going to be spun, pre-declaring them makes it appear simple to me.